src/lib/modules/snark/index.ts

```diff
@@ -0,0 +1,152 @@
+import { exec } from "child_process";
+import globule from "globule";
+import * as path from "path";
+import * as zkSnark from "snarkjs";
+import { CircuitsConfig, Embark } from "../../../typings/embark";
+import { EmbarkConfig } from "../../../typings/embarkConfig";
+import { Logger } from "../../../typings/logger";
+
+const fs = require("../../core/fs");
+
+zkSnark.bigInt.prototype.toJSON = function() { return this.toString(); };
+const circomBinary = path.join(__dirname, "../../../../node_modules/.bin/circom");
+const snarkjsBinary = path.join(__dirname, "../../../../node_modules/.bin/snarkjs");
+
+enum Extension {
+  Json = ".json",
+  Circom = ".circom",
+  VkProof = ".vk_proof",
+  VkVerifier = ".vk_verifier",
+  Solidity = ".sol",
+}
+
+export default class Snark {
+  private readonly embarkConfig: EmbarkConfig;
+  private readonly compiledCircuitsPath: string;
+  private readonly circuitsConfig: CircuitsConfig;
+  private readonly logger: Logger;
+
+  constructor(private readonly embark: Embark, private readonly options: object) {
+    this.logger = embark.logger;
+    this.circuitsConfig = embark.config.circuitsConfig;
+
+    this.embarkConfig = fs.readJSONSync(path.join(fs.dappPath(), "embark.json"));
+    this.compiledCircuitsPath = path.join(fs.dappPath(), ".embark", "snarks");
+
+    fs.ensureDirSync(this.compiledCircuitsPath);
+    this.embark.registerActionForEvent("build:beforeAll", this.compileAndGenerateContracts.bind(this));
+  }
+
+  private readonly compileAndGenerateContracts = async (callback: () => void) => {
+    try {
+      await Promise.all(this.compileCircuits());
+      await Promise.all(this.generateProofs());
+      this.addVerifiersToContracts();
+      callback();
+    } catch (error) {
+      console.error(error);
```

Comment:
You have access to the logger through `embark.logger`

Reply:
updated